src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -74,16 +74,29 @@ public UInt256 Hash
             sizeof(long) +  //NetworkFee
             sizeof(uint);   //ValidUntilBlock
 
-        public int Size => HeaderSize +
-            Attributes.GetVarSize() +   //Attributes
-            Cosigners.GetVarSize() +    //Cosigners
-            Script.GetVarSize() +       //Script
-            Witnesses.GetVarSize();     //Witnesses
+        /// <summary>
+        /// Assigned when tx is deserialized
+        /// </summary>
+        private int _size;
+        public int Size
+        {
+            get
+            {
+                return _size == 0 ? HeaderSize +
+                  Attributes.GetVarSize() +   //Attributes
+                  Cosigners.GetVarSize() +    //Cosigners
+                  Script.GetVarSize() +       //Script
+                  Witnesses.GetVarSize()      //Witnesses
+                  : _size;
+            }
+        }
 
         void ISerializable.Deserialize(BinaryReader reader)
         {
+            var startPosition = (int)reader.BaseStream.Position;
             DeserializeUnsigned(reader);
             Witnesses = reader.ReadSerializableArray<Witness>();
+            _size = (int)reader.BaseStream.Position - startPosition;
```

Comment:
BinaryReader could be not `seekable`, did you try with a `NetworkStream`?

Reply:
I think that is enough with caching the Size, but we can remove this change.

Reply:
Yes. It must be removed.